src/main/java/by/custom/utilcalculator/domain/UserProgress.java

```diff
@@ -8,50 +8,118 @@
 import java.util.Map;
 
 public class UserProgress implements Serializable {
-    private TransportType transportType = null;
+    private GeneralTransportType generalTransportType = null;
     private CountryOrigin countryOrigin = null;
     private OwnersType ownersType = null;
     private CarAge carAge = null;
-    private TypeOfEngine typeOfEngine = null;
-    private VolumeOfEngine volumeOfEngine = null;
+    private M1TypeOfEngine typeOfEngine = null;
+    private M1EngineVolume m1EngineVolume = null;
+    private ExceptM1TransportType exceptM1TransportType = null;
+    private N1N3TransportWeight transportWeightN1N2N3 = null;
+    private M2M3EngineType engineTypeM2M3 = null;
+    private M2EngineVolume m2EngineVolume = null;
+    private TruckUnitClass truckUnitClass = null;
+    private TruckUnitWeight truckUnitWeight = null;
+    private TrailerO4Type trailersO4Type = null;
     private final String chatID;
-    private Step currentQuestion;
+    private Step nextStep;
```

Comment:
Not sure that it's good decision store nextStep instead of current

Reply:
Highly likely you are right. It was completely my decision, I disobeyed your order to save here current step